src/communication/rpc_impl.rs

```diff
@@ -0,0 +1,486 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use async_trait::async_trait;
+use tokio::sync::oneshot::{Receiver, Sender};
+
+use crate::{
+    LocalUriProvider, UCode, UListener, UMessage, UMessageBuilder, UStatus, UTransport, UUri, UUID,
+};
+
+use super::{CallOptions, RpcClient, ServiceInvocationError, UPayload};
+
+fn handle_response_message(response: UMessage) -> Result<Option<UPayload>, ServiceInvocationError> {
+    if let Some(attribs) = response.attributes.as_ref() {
+        match attribs.commstatus.map(|v| v.enum_value_or_default()) {
+            Some(UCode::OK) | None => {
+                // successful invocation
+                response.payload.map_or(Ok(None), |payload| {
```

Comment:
May be more of a spec question, but given that I think you wrote or rewrote this part of the spec, I'm wondering why it's [considered non-mandatory](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uattributes.adoc#6-rpc-response-message-attributes) for a Response message to not contain a commstatus.

Seems like to me that's a fairly reasonable thing to require.